src/Traversals/bfs.jl

```diff
@@ -80,15 +81,50 @@ end
 end
 
 """
-    distances(g, s, alg=BFS())
-    distances(g, ss, alg=BFS())
+    distances(g, s, alg=BreadthFirst())
+    distances(g, ss, alg=BreadthFirst())
 
 Return a vector filled with the geodesic distances of vertices in  `g` from vertex `s` / unique vertices `ss`
-using BFS traversal algorithm `alg`.  For vertices in disconnected components the default distance is `typemax(T)`.
+using BreadthFirst traversal algorithm `alg`.  For vertices in disconnected components the default distance is `typemax(T)`.
 """
-function distances(g::AbstractGraph{T}, s, alg::BFS=BFS()) where T
+function distances(g::AbstractGraph{T}, s, alg::BreadthFirst=BreadthFirst()) where T
     d = fill(typemax(T), nv(g))
     state = DistanceState(d, one(T))
     traverse_graph!(g, s, alg, state)
     return state.distances
 end
+
+mutable struct PathState{T} <: AbstractTraversalState
+    u::T
+    v::T
+    exclude_vertices::Vector{T}
+    vertices_in_exclude::Bool # set to true if the source or dest are in excludes.
+end
+
+@inline function preinitfn!(s::PathState, visited)
+    visited[s.exclude_vertices] .= true
+    s.vertices_in_exclude = visited[s.u] | visited[s.v]
+    return !s.vertices_in_exclude
+end
+@inline newvisitfn!(s::PathState, u, v) = s.v != v 
```

Comment:
Is `@inline` necessary here? Also `u` is not used here, is that normal? If yes, it can be replaced by an underscore

Reply:
> Is @inline necessary here?
Not sure but I wanted to make it explicit since if it's not inlined then performance suffers./

> Also u is not used here, is that normal? If yes, it can be replaced by an underscore
I wanted to keep the function signatures the same as the prototypes.

Reply:
> Not sure but I wanted to make it explicit since if it's not inlined then performance suffers.

inlining perf is tricky, I usually let the compiler do the job of figuring out what to inline or not